api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceServiceImpl.java

```diff
@@ -181,8 +190,35 @@ public void deleteNotebookEntry(long workspaceId, long userId, String notebookPa
   }
 
   /** Deletes cohort entry from user_recently_modified_resource */
+  /**
+   * Since user_recently_modified_resource does not hold FK constraints, hence we have to explicitly
+   * delete COHORT REVIEWS and DATA SETs that are using the deleted cohort id
+   */
   @Override
   public void deleteCohortEntry(long workspaceId, long userId, long cohortId) {
+    // The logic to delete Cohort Review/DataSet used by deleted conceptSet is to be done
+    // for the new table (user_recently_modified_resource) only, hence execute it only if the
+    // feature flag enableDSCREntryInRecentModified is set to TRUE
+
+    if (workbenchConfigProvider.get().featureFlags.enableDSCREntryInRecentModified) {
+      List<Long> cohortReviewIds =
```

Comment:
Please confirm: we are doing this because when we delete a Cohort (the resource itself, not the RecentResource entry) we also delete all CRs and DSs which reference it.  Correct?

Please add comments describing this here, too.

Reply:
So workbench explicitly  does not delete CRs and DRs on deletion of COHORT,  the database does. I can update the comments

Reply:
makes sense, thanks